ls10-mel-memf/recipes-kernel/linux/files/mel-remote/0001-rpmsg_ls10_driver-add-tracing-support-for-MEL-as-rem.patch

```diff
@@ -0,0 +1,139 @@
+From 57c1062d120a5cf758e457c109d6009d0fe317d0 Mon Sep 17 00:00:00 2001
+From: Yasir-Khan <yasir_khan@mentor.com>
+Date: Mon, 4 Jul 2016 18:24:05 +0500
+Subject: [PATCH] rpmsg_ls10_driver: add tracing support for MEL as remote
+
+Signed-off-by: Yasir-Khan <yasir_khan@mentor.com>
+---
+ drivers/rpmsg/rpmsg_ls10_driver.c    | 30 ++++++++++++++++++++++++++++++
+ drivers/rpmsg/rpmsg_ls10_internals.h |  1 +
+ 2 files changed, 31 insertions(+)
+
+diff --git a/drivers/rpmsg/rpmsg_ls10_driver.c b/drivers/rpmsg/rpmsg_ls10_driver.c
+index 7b9e254..26d48b8 100644
+--- a/drivers/rpmsg/rpmsg_ls10_driver.c
++++ b/drivers/rpmsg/rpmsg_ls10_driver.c
+@@ -23,6 +23,10 @@
+  * GNU General Public License for more details.
+  */
+ 
++#include <trace/events/MEMF.h>
++#define CREATE_TRACE_POINTS
++#include <trace/events/Synchronization.h>
++
+ #include <linux/kernel.h>
+ #include <linux/init.h>
+ #include <linux/module.h>
+@@ -47,6 +51,9 @@
+ 
+ #include "rpmsg_ls10_internals.h"
+ 
++#define SUCCESS			0
++#define FAILURE			1
++
+ static DEFINE_IDA(rpmsg_ls10_dev_index);
+ 
+ /* Globals. */
+@@ -91,6 +98,8 @@ static void setup_shared_memory(void)
+ 
+ static bool ls10_rpmsg_virtio_notify(struct virtqueue *vq)
+ {
++	static bool executed = false;
++
+ 	/* Notify the other core. */
+ 	if (vq == ls10_rpmsg_p->vrings[0].vq)
+ 		/* Raise soft IRQ on GIC. */
+@@ -98,6 +107,11 @@ static bool ls10_rpmsg_virtio_notify(struct virtqueue *vq)
+ 	else
+ 		gic_raise_softirq_unicore(0, ls10_rpmsg_p->vring1);
+ 
++	if (!executed) {
++		executed = true;
++		trace_Synchronization_TriggerSend(100, 0, "Trigger");
++	}
++
+ 	return true;
+ }
+ 
+@@ -252,6 +266,8 @@ static void ipi_handler(void)
+ 
+ static int ls10_rpmsg_deinitialize(struct platform_device *pdev)
+ {
++	trace_MEMF_Remoteproc_Rsc_Deinit(ls10_rpmsg_p, SUCCESS);
++
+ 	unregister_virtio_device(&(ls10_rpmsg_p->virtio_dev));
+ 
+ 	put_device(&(ls10_rpmsg_p->mid_dev));
+@@ -269,12 +285,14 @@ static int ls10_rpmsg_initialize(struct platform_device *pdev)
+ 	int index;
+ 	struct virtio_device *virtio_dev;
+ 	int size;
++	int cpu_id = smp_processor_id();
```

Comment:
smp_processor_id doesn't seem appropriate here. 
Since MEL remote will always run on second core for this platform, kindly define cpu id and use that.